### PR TITLE
Bump GV version. Fix WebVR texture sharpness issues

### LINKED ISSUE #937: Poor rendering of detail in WebVR scenes
## Hardware
Oculus Go

## Steps to Reproduce
1. Load http://pano.yfway.com/30801042 or a demo from the [krapno's WebVR example](https://krpano.com/webvr/) in Firefox Reality.
2. Navigate to other panos by the gazing on a hotspot, and move your head around to notice the tracking is a bit slow.
4. Open the same site in Oculus Browser.
5. Upon navigating to other panos, notice the tracking is more stable than it is in Firefox Reality.

Originally reported in the [krpano forums](https://krpano.com/forum/wbb/index.php?page=Thread&postID=77699).

## Current Behavior
The tracking is laggy in [krpano-powered WebVR galleries](https://krpano.com/webvr/).

## Expected Behavior
The tracking is not laggy in [krpano-powered WebVR galleries](https://krpano.com/webvr/).

## Context
Originally reported in the [krpano forums](https://krpano.com/forum/wbb/index.php?page=Thread&postID=77699).

### PR DESCRIPTION
**Pull request by MortimerGoro**
Fixes #937

Do not merge yet, I created this to get some initial feedback of this solution. The problem is that the testcase creates a 2x canvas surface but we are still using the default size in our WebVR swapChain. This PR resizes the swapChain based on the received surface size (taking some thresholds into account).

We need a way to get the WebVR surface size received from Gecko in order to call `SetImmersiveSize` with the correct values. OpenGL ES doesn't have an API to get the texture size until 3.1. Anyway I don't like the OpenGL query solution (getters aren't good for performance). Ideally we can include the surface size in the external_vr struct or also in GeckoSurfaceTexture.java

### CODE REVIEW

**Comment by daoshengmu** [comment]
I tried to print the WebGL frambuffer size of this example [1] from Gecko. At the first creation, it tried to use `300x150`, but it failed, then finally using `1385x779`. The code tells me 
```
attemptSize.width /= 2;
attemptSize.height /= 2;
```

But, it got bigger numbers, I felt surprised. I need to trace the logic later.

At, `DeviceDelegateOculusVR::EnterVR()`, we use `renderWidth(1024) x renderHeight(1024)`, it uses the same buffer size that is shown at [2]. I would like to use the framebuffer size from WebGL to DeviceDelegateOculusVR::EnterVR() via ExternalVR, that should be a right fix.

[1] https://webvr.info/samples/04-simple-mirroring.html
[2] https://webvr.info/samples/00-hello-webvr.html

**Comment by MortimerGoro** [comment]
@daoshengmu 

I get 2048x1024 in the `04-simple-mirroring.html` webvr sample (1024x1024 per eye).

**Comment by MortimerGoro** [comment]
Related GV patch: https://bugzilla.mozilla.org/show_bug.cgi?id=1554626

**Comment by MortimerGoro** [comment]
GV patch already merged and updated in the PR. 

This is ready for the final review

**Comment by bluemarvin** [comment]
@MortimerGoro Wouldn't this be an issue for WaveVR and GoogleVR too?

**Comment by MortimerGoro** [comment]
@bluemarvin true, I forgot about that, I'll update the wave and google backends

**Comment by daoshengmu** [comment]
It looks good to me on OculusVR, I will review it again once a new update comes for other platforms.

**Comment by MortimerGoro** [comment]
Wave and Daydream platforms are ready

File: app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp [hunk 1]
```diff
@@ -1110,6 +1110,30 @@ DeviceDelegateOculusVR::RegisterImmersiveDisplay(ImmersiveDisplayPtr aDisplay) {
...
+  m.GetImmersiveRenderSize(recommendedWidth, recommendedHeight);
+
+  const uint32_t maxWidth = recommendedWidth * 2;
+  const uint32_t maxHeight = recommendedHeight * 2;
+  const uint32_t minWidth = recommendedWidth / 2;
+  const uint32_t minHeight = recommendedHeight / 2;
+
+  const uint32_t targetWidth = (uint32_t) fmaxf(fminf(aEyeWidth, maxWidth), minWidth);
+  const uint32_t targetHeight = (uint32_t) fmaxf(fminf(aEyeHeight, maxHeight), minHeight);
+
```
Comment (daoshengmu) [inline app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: nit: I am thinking if we can move 
```
   const uint32_t maxWidth = recommendedWidth * 2;
   const uint32_t maxHeight = recommendedHeight * 2;
   const uint32_t minWidth = recommendedWidth / 2;
   const uint32_t minHeight = recommendedHeight / 2;
   const uint32_t targetWidth = (uint32_t) fmaxf(fminf(aEyeWidth, maxWidth), minWidth);
   const uint32_t targetHeight = (uint32_t) fmaxf(fminf(aEyeHeight, maxHeight), minHeight);
```
to class DeviceDelegate as `GetTargetWidth()` and `GetTargetHeight()`, And getting them via `uint32 w = GetTargetWidth(recommendedWidth);`, `uint32 h = GetTargetHeight(recommendedHeight);`  how do you think about it ?

Comment (MortimerGoro) [reply app/src/oculusvr/cpp/DeviceDelegateOculusVR.cpp]: Done

**Comment by bluemarvin** [comment]
@MortimerGoro What pages did you use to test this?

**Comment by bluemarvin** [comment]
Specifically, using this site: https://www.tourdeforce360.com/grandtheater/vtour.html

It is 100% reproducible.

**Comment by MortimerGoro** [comment]
@bluemarvin PR updated with the fix for daydream

**Pull request merged**